### PR TITLE
Mko/redux-profile-update-getprofile

### PR DESCRIPTION
**Pull request by mko5118**
### Description

Connect get user profile endpoint with front end to allow for viewing of individual user's profile pages

### Changes

REDUX
- Update profile.types.js -- GET_PROFILE, PROFILE_ERROR
- Update profile.actions.js -- getUserProfile
- Update profile.reducer.js -- getUserProfile

PROFILE COMPONENT
- Add profile folder to components-page folder
- Add Profile.js -- component logic
- Add profile.module.scss -- component styling

PROFILE PAGE COMPONENT
- Update ProfilePage.js -- import Profile component

### Verification

image

### Trello Link

https://trello.com/c/TUm67dyR/24-hook-up-frontend-backend-routtes

### Notes

- Styling is not complete, general layout has been implemented but further tweaking of styling is needed (ie. better mobile responsiveness of avatar image, font sizings, etc)
- The Profile component will likely be broken up into smaller components (the logic for the postsContainer section will for sure be broken into a separate component and imported into Profile.js)
- If viewing your own profile, Edit Profile Button will render, if not a Friend Request Button will render, further tweaking of whether to display button at all if user is already friends with the current profile will be determined
- Edit Button / Friend Request button is not functional currently
- Comment link to view comments is not functional currently
- Friends Count is only for view, no actions enabled to click

### CODE REVIEW

File: client/src/components-page/profile/Profile.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+const Profile = ({ currentProfile, loading, auth, match, getUserProfile }) => {
+  // 'match' passed down as prop from AppRouter
+  useEffect(() => {
+    getUserProfile(match.params.id);
+  }, [getUserProfile, match]);
+
+  return (
+    <div className={style.profile}>
+    {
+      currentProfile === null || loading
```
Comment (Formless) [inline client/src/components-page/profile/Profile.js]: I think this is written a bit awkwardly. Consider returning the div with the spinner in it before line 20, based on the conditional.

e.g. 
```
if (currentProfile === null || loading)
  return ( // stuff
else
  return ( // the profile
```

You will end up duplicating a bit but I think it reads cleaner.

File: client/src/components-page/profile/profile.module.scss [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+      .avatar {
+        height: 100%;
+        margin-right: 1rem;
+      }
+
+      .bioContainer {
+
+        .bioHeader {
+
```
Comment (Formless) [inline client/src/components-page/profile/profile.module.scss]: I assume this is to be filled out later?

File: client/src/components-page/profile/Profile.js (hunk 1 above)
Comment (mko5118) [reply client/src/components-page/profile/Profile.js]: True yeah I agree, I'll take a look at that tomorrow

File: client/src/components-page/profile/profile.module.scss (hunk 2 above)
Comment (mko5118) [reply client/src/components-page/profile/profile.module.scss]: Yeah, there's some styling I'll be adding in there not sure what at the moment so it's currently a placeholder

**Pull request merged**